src/Common/src/System/Globalization/IdnMapping.Windows.cs

```diff
@@ -20,14 +20,22 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             }
 
             // Do the conversion
-            char[] output = new char[length];
-            length = Interop.mincore.IdnToAscii(flags, unicode, count, output, length);
-            if (length == 0)
+            const int StackAllocThreshold = 512; // arbitrary limit to switch from stack to heap allocation
+            if (length < StackAllocThreshold)
             {
-                ThrowForZeroLength("unicode", SR.Argument_IdnIllegalName, SR.Argument_InvalidCharSequenceNoIndex);
+                char* output = stackalloc char[length];
+                Interop.mincore.IdnToAscii(flags, unicode, count, output, length);
```

Comment:
No, that's different.  IdnToAscii and IdnToUnicode return a new length that needs to be both used and checked.